org.scala-ide.sdt.core/src/org/scalaide/refactoring/internal/move/MoveClass.scala

```diff
@@ -131,6 +131,7 @@ class MoveClass extends RefactoringExecutorWithWizard {
             (change :: textChanges, newFiles)
           case (change: NewFileChange, (textChanges, newFilesChanges)) =>
             (textChanges, change :: newFilesChanges)
+          case (other, _) => throw new AssertionError(s"Unexpected change $other")
```

Comment:
Please improve the message in a way that it explains that only `TextChange` and `NewFileChange` make sense here.